src/CommunityToolkit.Maui.UnitTests/Views/GravatarImageSource/GravatarImageSourceTests.cs

```diff
@@ -0,0 +1,463 @@
+	}
+
+	[Fact]
+	public void TestControlImageButton()
+	{
+		ImageButton testControl = new()
+		{
+			Source = new GravatarImageSource(),
+		};
+		Assert.True(testControl.Source is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/?s=80"), ((GravatarImageSource)testControl.Source).Uri);
+	}
+
+	[Fact]
+	public void TestControlImageButtonWithEmail()
+	{
+		ImageButton testControl = new()
+		{
+			Source = new GravatarImageSource() { Email = email },
+		};
+		Assert.True(testControl.Source is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=80&d=mp"), ((GravatarImageSource)testControl.Source).Uri);
+	}
+
+	[Fact]
+	public void TestControlImageButtonWithEmailandSize()
+	{
+		ImageButton testControl = new()
+		{
+			Source = new GravatarImageSource() { Email = email },
+			WidthRequest = 72,
+			HeightRequest = 73
+		};
+		Assert.True(testControl.Source is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=80&d=mp"), ((GravatarImageSource)testControl.Source).Uri);
+	}
+
+	[Fact]
+	public void TestControlImageCell()
+	{
+		ImageCell testControl = new()
+		{
+			ImageSource = new GravatarImageSource(),
+		};
+		Assert.True(testControl.ImageSource is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/"), ((GravatarImageSource)testControl.ImageSource).Uri);
+	}
+
+	[Fact]
+	public void TestControlImageCellWithEmail()
+	{
+		ImageCell testControl = new()
+		{
+			ImageSource = new GravatarImageSource() { Email = email },
+		};
+		Assert.True(testControl.ImageSource is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=0&d=mp"), ((GravatarImageSource)testControl.ImageSource).Uri);
+	}
+
+	[Fact]
+	public void TestControlImageWithEmail()
+	{
+		Image testImage = new()
+		{
+			Source = new GravatarImageSource() { Email = email }
+		};
+		Assert.True(testImage.Source is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=80&d=mp"), ((GravatarImageSource)testImage.Source).Uri);
+	}
+
+	[Fact]
+	public void TestControlMenuItem()
+	{
+		MenuItem testControl = new()
+		{
+			IconImageSource = new GravatarImageSource(),
+		};
+		Assert.True(testControl.IconImageSource is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/"), ((GravatarImageSource)testControl.IconImageSource).Uri);
+	}
+
+	[Fact]
+	public void TestControlMenuItemControlWithEmail()
+	{
+		MenuItem testControl = new()
+		{
+			IconImageSource = new GravatarImageSource() { Email = email },
+		};
+		Assert.True(testControl.IconImageSource is GravatarImageSource);
+		Assert.Equal(new Uri("https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=0&d=mp"), ((GravatarImageSource)testControl.IconImageSource).Uri);
+	}
+
+	[Fact]
+	public void TestControlPage()
```

Comment:
I can't tell what these tests verify, sure they confirm that what we set stays set on the control but do they confirm that the gravatar image is rendered? I wonder if we might be able to remove the need for these and run some tests to verify that `GravatarImageSource` plays well with `UriImageSourceService` so that it will perform the correct loading behaviour. I am busy for the next couple of days but I will try and have a play with this concept after that.

Reply:
I thought it best to do a belt-n-braces approach and at least perform 'some' tests for any control that could use ImageSource.